### PR TITLE
add atomixrex

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
I have an error with the real time library: 
```
/home/conda/feedstock_root/build_artifacts/atomicrex_1570366502535/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: CMakeFiles/test_minimizer.cpp.dir/tests/minimizers/test_minimizer.cpp.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'
/home/conda/feedstock_root/build_artifacts/atomicrex_1570366502535/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/atomicrex_1570366502535/_build_env/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/librt.so.1: error adding symbols: DSO missing from command line
```

**Comment by jan-janssen** [comment]
It seems like I need glibc 2.17 https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=6e6249d0b461b952d0f544792372663feb6d792a;ds=sidebyside - but 2.2.5 is available via conda-forge.

File: recipes/atomicrex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake 
+    - make
+    - python
```
Comment (isuruf) [inline recipes/atomicrex/meta.yaml]: `python` should be in `host`

File: recipes/atomicrex/disabletests.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+disable tests as they require glibc 2.17
```
Comment (isuruf) [inline recipes/atomicrex/disabletests.patch]: Instead of disabling tests try doing `export LDFLAGS="$LDFLAGS -lrt"`

Comment (jan-janssen) [reply recipes/atomicrex/disabletests.patch]: Unfortunately that still gives the same error: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=78216&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c

Comment (isuruf) [reply recipes/atomicrex/disabletests.patch]: Can you run `make VERBOSE=1` for more verbosity?

File: recipes/atomicrex/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/atomicrex/meta.yaml]: The issue is they use the location of `cmake` to find the python executable. I know it is a bit strange and I am in contact with the developers to fix it, but for now I thought that adding python to the `build` section could solve this issue: https://gitlab.com/atomicrex/atomicrex/blob/master/cmake/Testing.cmake#L43

Comment (isuruf) [reply recipes/atomicrex/meta.yaml]: I think you are misreading that line. CMAKE_BINARY_DIR is not where cmake is. It is the build directory and `${CMAKE_BINARY_DIR}/python` isn't for the python executable, it is the path to the python files which are in this directory called python

**Comment by jan-janssen** [comment]
The issue seems to be in https://cmake.org/cmake/help/v3.0/module/FindPythonInterp.html - because this returns `/opt/conda/bin/python3.7` instead of `${PREFIX}/bin/python3.7`

**Comment by isuruf** [comment]
See https://github.com/conda-forge/cmake-feedstock/issues/91 for some workarounds

**Comment by jan-janssen** [comment]
FindPythonInterp relies on Python_ADDITIONAL_VERSIONS which is set in https://gitlab.com/atomicrex/atomicrex/blob/master/src/3rdparty/pybind11/tools/pybind11Tools.cmake#L15 - so I can not overwrite it easily. So for now I only build the Python 3.7 version.

File: recipes/atomicrex/disabletests.patch (hunk 2 above)
Comment (jan-janssen) [reply recipes/atomicrex/disabletests.patch]: Then I get: 
```
$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++  -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/atomicrex-1.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -O3 -DNDEBUG  -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -lrt -rdynamic CMakeFiles/test_minimizer.cpp.dir/tests/minimizers/test_minimizer.cpp.o  -o test_minimizer.cpp src/arexlib/libatomicrexlib.a $PREFIX/lib/libboost_filesystem.a -lxml2 $PREFIX/lib/libmuparser.so -lnlopt -fopenmp -lgfortran -lgomp -lquadmath -lpthread 
/home/janssen/miniconda3/conda-bld/atomicrex_1570396329992/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: CMakeFiles/test_minimizer.cpp.dir/tests/minimizers/test_minimizer.cpp.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'
/home/janssen/miniconda3/conda-bld/atomicrex_1570396329992/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /home/janssen/miniconda3/conda-bld/atomicrex_1570396329992/_build_env/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/../lib/librt.so: error adding symbols: DSO missing from command line
```

So to me it seems disabling the tests is the easiest way for now.

Comment (isuruf) [reply recipes/atomicrex/disabletests.patch]: Then you can add, `"-lrt"` to https://gitlab.com/atomicrex/atomicrex/blob/master/cmake/Testing.cmake#L61 by patching.

**Comment by isuruf** [comment]
Can you also make this library link to boost dynamically instead of statically?

**Comment by jan-janssen** [comment]
Sorry, I do know how to switch to linking boost dynamically. ...

**Comment by isuruf** [comment]
Try adding `-DBoost_USE_STATIC_LIBS=OFF` (Can you keep the verbose build so that I can see what's going on?)

**Comment by jan-janssen** [comment]
Then I get the following error message:

```
CMake Error at /home/conda/feedstock_root/build_artifacts/atomicrex_1570401529466/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/cmake/Boost-1.70.0/BoostConfig.cmake:95 (find_package):
  Found package configuration file:

    /home/conda/feedstock_root/build_artifacts/atomicrex_1570401529466/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/cmake/boost_filesystem-1.70.0/boost_filesystem-config.cmake

  but it set boost_filesystem_FOUND to FALSE so package "boost_filesystem" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.
```

**Comment by isuruf** [comment]
Try https://github.com/conda-forge/boost-cpp-feedstock/issues/48#issuecomment-499621074

File: recipes/atomicrex/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - make
+    - cmake 
+  host:
+    - python
+    - boost
```
Comment (isuruf) [inline recipes/atomicrex/meta.yaml]: Change this to `boost-cpp`

File: recipes/atomicrex/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - python
+    - boost 
+    - libxml2
+    - muparser
+    - nlopt
+    - vim
+    - pybind11
+  run:
+    - python
+    - boost
```
Comment (isuruf) [inline recipes/atomicrex/meta.yaml]: Same here

File: recipes/atomicrex/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - vim
+    - pybind11
+  run:
+    - python
+    - boost 
+    - libxml2
+    - muparser
+    - nlopt
+    - vim
+    - pybind11
```
Comment (isuruf) [inline recipes/atomicrex/meta.yaml]: Why is this needed in `run`?

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/atomicrex/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    - {{ compiler('fortran') }}
+    - make
+    - cmake 
+  host:
+    - python
+    - boost-cpp
+    - libxml2
+    - muparser
+    - nlopt
+    - vim
```
Comment (chrisburr) [inline recipes/atomicrex/meta.yaml]: Just because I haven’t seen it before, is this needed in host? Glancing at the source it looks like it is used as part of the build system and should be in build?
